core/jobs/batch_jobs/suggestion_stats_computation_jobs.py

```diff
@@ -178,17 +393,28 @@ def _generate_stats(
                 content_word_count: int. The word count of the content of
                     the suggestion.
                 last_updated_date: str. When was the suggestion last updated.
+                created_date: str. When was the suggestion created.
         """
         # When opportunity is not available we leave the topic ID empty.
         topic_id = ''
         if opportunity is not None:
             topic_id = opportunity.topic_id
 
         for suggestion in suggestions:
-            key = (
-                suggestion_models
-                .TranslationContributionStatsModel.construct_id(
-                    suggestion.language_code, suggestion.author_id, topic_id))
+            if translation_contribution_key_required:
+                key = (
+                    suggestion_models
+                    .TranslationContributionStatsModel.construct_id(
+                        suggestion.language_code, suggestion.author_id, topic_id))
+            else:
+                key = (
+                    suggestion_models
+                    .TranslationReviewStatsModel.construct_id(
+                        suggestion.language_code,
+                        suggestion.final_reviewer_id,
+                        topic_id
+                    )
+                )
```

Comment:
I believe this can be simplified to one line if we pass in the model reference instead of a boolean. Ditto for the question stats method.